dot/peerset/peerset_test.go

```diff
@@ -25,96 +29,87 @@ func TestPeerSetBanned(t *testing.T) {
+	processor.EXPECT().Process(Message{Status: Drop, setID: 0x0, PeerID: "testPeer1"})
 	handler.ReportPeer(rpc, peer1)
 	time.Sleep(time.Millisecond * 100)
 
-	checkMessageStatus(t, <-ps.resultMsgCh, Drop)
-
 	// check that an incoming connection from that node gets refused.
+	processor.EXPECT().Process(Message{Status: Reject, setID: 0x0, PeerID: "testPeer1"})
 	handler.Incoming(0, peer1)
-	checkMessageStatus(t, <-ps.resultMsgCh, Reject)
 
 	// wait a bit for the node's reputation to go above the threshold.
 	time.Sleep(time.Millisecond * 1200)
 
 	// try again. This time the node should be accepted.
+	processor.EXPECT().Process(Message{Status: Accept, setID: 0x0, PeerID: "testPeer1"})
 	handler.Incoming(0, peer1)
-	require.NoError(t, err)
-	checkMessageStatus(t, <-ps.resultMsgCh, Accept)
 }
 
 func TestAddReservedPeers(t *testing.T) {
 	t.Parallel()
 
-	handler := newTestPeerSet(t, 0, 2, []peer.ID{bootNode}, []peer.ID{}, false)
+	ctrl := gomock.NewController(t)
+	processor := NewMockMessageProcessor(ctrl)
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: bootNode})
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: reservedPeer})
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: reservedPeer2})
+
+	handler := newTestPeerSet(t, 0, 2, []peer.ID{bootNode}, []peer.ID{}, false, processor)
 	ps := handler.peerSet
 
 	handler.AddReservedPeer(0, reservedPeer)
 	handler.AddReservedPeer(0, reservedPeer2)
 
 	time.Sleep(time.Millisecond * 200)
 
-	expectedMsgs := []Message{
-		{Status: Connect, setID: 0, PeerID: bootNode},
-		{Status: Connect, setID: 0, PeerID: reservedPeer},
-		{Status: Connect, setID: 0, PeerID: reservedPeer2},
-	}
-
 	require.Equal(t, uint32(1), ps.peerState.sets[0].numOut)
-	require.Equal(t, 3, len(ps.resultMsgCh))
-
-	for i := 0; ; i++ {
-		if len(ps.resultMsgCh) == 0 {
-			break
-		}
-		msg := <-ps.resultMsgCh
-		require.Equal(t, expectedMsgs[i], msg)
-	}
 }
 
 func TestPeerSetIncoming(t *testing.T) {
 	t.Parallel()
 
-	handler := newTestPeerSet(t, 2, 1, []peer.ID{bootNode}, []peer.ID{}, false)
-	ps := handler.peerSet
+	ctrl := gomock.NewController(t)
+	processor := NewMockMessageProcessor(ctrl)
 
-	// connect message will be added ingoing queue for bootnode.
-	checkMessageStatus(t, <-ps.resultMsgCh, Connect)
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: bootNode})
+	handler := newTestPeerSet(t, 2, 1, []peer.ID{bootNode},
+		[]peer.ID{}, false, processor)
 
+	processor.EXPECT().Process(Message{Status: Accept, setID: 0, PeerID: incomingPeer})
 	handler.Incoming(0, incomingPeer)
-	checkMessageStatus(t, <-ps.resultMsgCh, Accept)
 
+	processor.EXPECT().Process(Message{Status: Accept, setID: 0, PeerID: incoming2})
 	handler.Incoming(0, incoming2)
-	checkMessageStatus(t, <-ps.resultMsgCh, Accept)
 
+	processor.EXPECT().Process(Message{Status: Reject, setID: 0, PeerID: incoming3})
 	handler.Incoming(0, incoming3)
```

Comment:
put assert/require to show impact of handler.incoming (here and at other places in this test)